packages/host/app/components/operator-mode/container.gts

```diff
@@ -611,38 +614,60 @@ export default class OperatorModeContainer extends Component<Signature> {
     this.operatorModeStateService.updateSubmode(submode);
   }
 
-  @action resetCodeModeCardError() {
-    this.codeModeCardError = null;
-  }
-
-  @action loadCardForCodeMode(codePath: URL) {
-    this.loadCardForCodeModeTask.perform(codePath);
+  @action resetLoadFileError() {
+    this.loadFileError = null;
   }
 
-  loadCardForCodeModeTask = restartableTask(async (codePath: URL) => {
-    try {
-      let realmURL = this.cardService.getRealmURLFor(codePath);
-      if (!realmURL) {
-        throw new Error('URL is not in any realms');
-      }
-
-      this.codeModeCard = await this.cardService.loadModel(codePath);
-      this.operatorModeStateService.updateCodePath(codePath);
-    } catch (e: any) {
-      this.codeModeCardError =
-        e.message === 'URL is not in any realms'
-          ? e.message
-          : 'File is not found';
+  openFile = maybe(this, (context) => {
+    let realmURL = this.cardService.getRealmURLFor(this.codePath);
+    if (!realmURL) {
+      return undefined;
+    }
+    const relativePath = this.codePath
+      .toString()
+      .replace(realmURL.toString(), '');
+    console.log(relativePath);
+    if (relativePath) {
+      return file(context, () => ({
+        relativePath,
+        // @ts-ignore type 'undefined' is not assignable to type 'string | URL'
+        // we have ensured its value above.
+        realmURL: new RealmPaths(realmURL).url,
+        onStateChange: (state) => {
+          if (state === 'not-found') {
+            this.loadFileError = 'File is not found';
+          }
+        },
+      }));
+    } else {
+      return undefined;
     }
   });
 
   get codePath() {
     return (
       this.operatorModeStateService.state.codePath ??
-      new URL(this.lastCardInRightMostStack.id)
+      new URL(this.lastCardInRightMostStack.id + '.json')
```

Comment:
oh wait why is this needed?

Reply:
For one. this.lastCardInRightMostStack types are wrong. It doesn't guarantee the CardDef return. During runtime, theres a case where the right most card doesn't exist


https://github.com/cardstack/boxel/assets/8165111/a761a414-ef72-4bc0-963f-f99ec541c2de

Reply:
Id favor just hardcoding the index card in there